.travis.yml

```diff
@@ -1,8 +1,36 @@
+sudo: false
 language: node_js
-node_js:
-  - "6"
-before_script:
-  - npm install -g elm elm-test elm-doc-test
-  - elm-test init
-  - elm-doc-test
-script: elm-test tests/Doc/Main.elm
+node_js: "node"
+os: linux
+env: ELM_VERSION=0.18.0
+
+cache:
+  directories:
+    - test/elm-stuff/build-artifacts
+    - sysconfcpus
+
+  before_install:
+    - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+    - | # epic build time improvement - see https://github.com/elm-lang/elm-compiler/issues/1473#issuecomment-245704142
+      if [ ! -d sysconfcpus/bin ];
+      then
+      git clone https://github.com/obmarg/libsysconfcpus.git;
```

Comment:
I don't really like having the travis tests depending of an external github repo used only here...

Reply:
I totally agree having the TravisCI tests depending on an external repo is not preferred. The problem I had was TravisCI tests were taking extremely long to run, 18 min+. This `travis.yml`  is from the popular elm package [elm-community/json-extra](https://github.com/elm-community/json-extra/blob/master/.travis.yml) . Maybe there is a simpler way.

Reply:
Here are some travis.yml from other elm packages
+ [elm-community/elm-time](https://github.com/elm-community/elm-time/blob/master/.travis.yml)
+ [elm-community/graph](https://github.com/elm-community/graph/blob/master/.travis.yml)
+ [1602/json-schema](https://github.com/1602/json-schema/blob/4.1.0/.travis.yml)
+ [1602/json-value](https://github.com/1602/json-value/blob/1.1.0/.travis.yml)
+ [Bogdanp/elm-ast](https://github.com/Bogdanp/elm-ast/blob/8.0.12/.travis.yml)

All pretty complicated unfortunately, seems to be the trend.